tests/rptest/tests/tx_admin_api_test.py

```diff
@@ -0,0 +1,166 @@
+# Copyright 2021 Vectorized, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from ducktape.mark.resource import cluster
+from rptest.clients.types import TopicSpec
+from rptest.services.admin import Admin
+
+import confluent_kafka as ck
+
+import random
+import time
+
+from rptest.tests.redpanda_test import RedpandaTest
+
+
+class TxAdminTest(RedpandaTest):
+    topics = (TopicSpec(name="tx_test",
+                        partition_count=3,
+                        replication_factor=3), )
+
+    def __init__(self, test_context):
+        super(TxAdminTest,
+              self).__init__(test_context=test_context,
+                             num_brokers=3,
+                             extra_rp_conf={
+                                 "enable_idempotence":
+                                 True,
+                                 "enable_transactions":
+                                 True,
+                                 "tx_timeout_delay_ms":
+                                 10000000,
+                                 "abort_timed_out_transactions_interval_ms":
+                                 10000000
+                             })
+
+        self.admin = Admin(self.redpanda)
+
+    @cluster(num_nodes=3)
+    def test_simple_get_transaction(self):
+        self.producer1 = ck.Producer({
+            'bootstrap.servers':
+            self.redpanda.brokers(),
+            'transactional.id':
+            '0',
+        })
+        self.producer2 = ck.Producer({
+            'bootstrap.servers':
+            self.redpanda.brokers(),
+            'transactional.id':
+            '1',
+        })
+        self.producer1.init_transactions()
+        self.producer2.init_transactions()
+        self.producer1.begin_transaction()
+        self.producer2.begin_transaction()
+
+        for topic in self.topics:
+            for partition in range(topic.partition_count):
+                self.producer1.produce(topic.name, '0', '0', partition)
+                self.producer2.produce(topic.name, '0', '1', partition)
+
+        self.producer1.flush()
+        self.producer2.flush()
+
+        pid1 = -1
+        epoch1 = -1
+        pid2 = -1
+        epoch2 = -1
+
+        for topic in self.topics:
+            for partition in range(topic.partition_count):
+                txs_info = self.admin.get_transactions(topic.name, partition)
+
+                if pid1 == -1:
+                    pid1 = txs_info['active_transactions'][0]['producer_id'][
```

Comment:
The problem here is that the test doesn't control for cardinality (if the api returns more active transactions than we expect then the test won't catch it) and depends on order of the active transacitons while there is no expectations that we should preserve it.

It should look something like:

```python
expected_pids = None

def extract_pid(tx):
    return (tx["producer_id"]["id"], tx["producer_id"]["epoch"])

for topic in ...:
  for partition in ...:
    txs_info = ...
    assert len(txs_info['expired_transactions']) == 0
    if expected_pids == None:
      expected_pids = set(map(extract_pid, txs_info['active_transactions']))
    assert len(expected_pids) == len(txs_info['active_transactions'])
    # check that all active_transactions are in expected_pids
      ...
```